src/lib/core/datetime.c

```diff
@@ -143,3 +143,149 @@ datetime_compare(const struct datetime *lhs, const struct datetime *rhs)
 
 	return COMPARE_RESULT(lhs->nsec, rhs->nsec);
 }
+
+/**
+ * Interval support functions: stringization and operations
+ */
+
+#define NANOS_PER_SEC 1000000000
+
+static inline int64_t
+interval_seconds(const struct datetime_interval *ival)
+{
+	return (int64_t)ival->sec + ival->nsec / NANOS_PER_SEC;
+}
+
+static inline int
+interval_days(const struct datetime_interval *ival)
+{
+	return interval_seconds(ival) / SECS_PER_DAY;
+}
+
+static inline int
+interval_hours(const struct datetime_interval *ival)
+{
+	return interval_seconds(ival) / (60 * 60);
```

Comment:
Probably, because everybody could easily recognize hardcoded number of seconds per hour or minute, but hardly could remember number of seconds per day. (There are number of similar constants spread over `datetime.c` here and there and nobody complained yet, because they are used in obvious contexts)